cmd/erasure-metadata.go

```diff
@@ -205,10 +205,14 @@ func (fi FileInfo) TransitionInfoEquals(ofi FileInfo) bool {
 
 // MetadataEquals returns true if FileInfos Metadata maps are equal, false otherwise.
 func (fi FileInfo) MetadataEquals(ofi FileInfo) bool {
-	if len(fi.Metadata) != len(ofi.Metadata) {
-		return false
-	}
 	for k, v := range fi.Metadata {
+		if k == reservedMetadataPrefixLowerDataShardFix {
```

Comment:
This is problematic, since it will affect quorum calculations for listing at least in [xlMetaV2Object signature](https://github.com/minio/minio/blob/81e43b87c2b9eb9a112e21aaaea04a202c4f4d55/cmd/xl-storage-format-v2.go#L520).

If we want to update metadata it should be done for *all* disks, otherwise we will risk running into all kinds of trouble. Not doing that is a rather high risk change.